src/machine.h

```diff
@@ -190,6 +190,7 @@ typedef struct VirtMachine {
     uint64_t    trace;
 
     /* For co-simulation only, they are -1 if nothing is pending. */
+    bool        cosim;
```

Comment:
The comment doesn't quite make sense now; the fields were only for cosim so cosim would be redundant. Can you remind me why you need this?  Co-simulation of Interrupts is the trickies part and I expect that many clients might need to customize this. Maybe we could refactor this in a way that keeps the special sauce in each client.

Reply:
The cosim flag qualifies if we need to look at the pending interrupts. The bug in the interpreter loop was that when in standalone mode it always enforced additional checks that would not be set as there is no DUT to handle interrupts. Maybe this could indeed be refactored differently.